<?xml version="1.0"?>
<launch>

    <arg name="kitting_urdf_xacro"/>
    <arg name="limited"/>

    <param name="robot_description"
      command="$(find xacro)/xacro &#x002D;&#x002D;inorder '$(arg kitting_urdf_xacro)'
      hardware_interface:=EffortJointInterface
      use_effort_interface:=true
      joint_limited:=$(arg limited)"/>

    <rosparam command="load" file="$(find joint_state_controller)/joint_state_controller.yaml"/>
    <node name="joint_state_controller_spawner" pkg="controller_manager" type="spawner" output="screen" args="joint_state_controller"/>


    <!-- Publish robot to tf tree -->
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher"/>

  <!-- push robot_description to factory and spawn robot in gazebo -->
  <node name="kitting_spawner" pkg="gazebo_ros" type="spawn_model"
        args="
          -urdf
          -param robot_description
          -J linear_arm_actuator_joint 0
          -J shoulder_pan_joint 0
          -J shoulder_lift_joint -1.25
          -J elbow_joint 1.74
          -J wrist_1_joint -2.66
          -J wrist_2_joint -1.51
          -J wrist_3_joint 0
          -model kitting"
        respawn="false" output="screen">
  </node>

  <!-- Undo this and use files from ur_gazebo when changes merged upstream. -->
  <include file="$(find nist_gear)/robots/ur10/ur_gazebo/launch/controller_utils.launch"/>

  <rosparam file="$(find nist_gear)/robots/ur10/ur_description/controller/kitting_arm_controller_ur10_custom.yaml" 
  ns="/ariac/kitting" 
  command="load"/>
      <!-- load controllers-->
    <node name="kitting_controller_spawner" pkg="controller_manager" type="spawner" respawn="false"
      output="screen" ns="/ariac/kitting" args="--namespace=/ariac/kitting kitting_arm_controller"/>
</launch>


